crates/proof-of-sql-parser/src/posql_time/unit.rs

```diff
@@ -53,10 +67,78 @@ impl fmt::Display for PoSQLTimeUnit {
 #[cfg(test)]
 #[allow(deprecated)]
 mod time_unit_tests {
-
-    use crate::posql_time::{timestamp::PoSQLTimestamp, unit::PoSQLTimeUnit};
+    use super::*;
+    use crate::{error::PoSQLTimestampError, posql_time::timestamp::PoSQLTimestamp};
     use chrono::{TimeZone, Utc};
 
+    #[test]
+    fn test_valid_precisions() {
+        assert_eq!(PoSQLTimeUnit::try_from("0"), Ok(PoSQLTimeUnit::Second));
+        assert_eq!(PoSQLTimeUnit::try_from("3"), Ok(PoSQLTimeUnit::Millisecond));
+        assert_eq!(PoSQLTimeUnit::try_from("6"), Ok(PoSQLTimeUnit::Microsecond));
+        assert_eq!(PoSQLTimeUnit::try_from("9"), Ok(PoSQLTimeUnit::Nanosecond));
+    }
+
+    #[test]
+    fn test_invalid_precisions() {
+        // Test some random incorrect values
+        assert!(
+            matches!(PoSQLTimeUnit::try_from("-1"), Err(PoSQLTimestampError::UnsupportedPrecision(x)) if x == "-1")
+        );
```

Comment:
You don't really need the `if x == "-1"` clause. Usually we do not do assert_eq on error since error message changes shouldn't normally trigger test changes. However since `x` in `PoSQLTimestampError::UnsupportedPrecision(x)` is entirely dependent on the invalid precision we may as well have `assert_eq!` instead.

Reply:
you asked me to add a negative number case below

Reply:
Yes. Please see below.